src/publisher_registry/publisher_registry.cairo

```diff
@@ -441,5 +443,26 @@ mod PublisherRegistry {
         // gas::withdraw_gas_all(get_builtin_costs()).expect('Out of gas');
         _iter_publisher_sources(self, cur_idx + 1_usize, max_idx, publisher, ref sources_arr)
     }
+
+    // @notice check if a given contract address is already associated to a publisher 
+    // @param address: address to check
+    // @returns a boolean 
+    fn is_address_registered(self: @ContractState, address: ContractAddress) -> bool {
+        let mut cur_idx = 0;
+        let arr_len = self.publishers_storage_len.read();
+        let mut boolean = false;
+        loop {
+            if (cur_idx == arr_len) {
+                break ();
+            }
+            let publisher = self.publishers_storage.read(cur_idx);
+            let publisher_address = self.publisher_address_storage.read(publisher);
+            if (publisher_address == address) {
+                boolean = true;
```

Comment:
Just return directly here

Reply:
afaik we cannot return directly from a loop